Makefile

```diff
@@ -114,521 +105,312 @@ endef
+
+$(ASMDIFFER_APP): | $(VENV_DIR)
+	git submodule update --init $(dir $(ASMDIFFER_APP))
+$(M2C_APP): tools/python-dependencies.make.chkpt | $(VENV_DIR)
+	git submodule update --init $(dir $(M2C_APP))
+$(PERMUTER_APP): | $(VENV_DIR)
+	git submodule update --init $(dir $(PERMUTER_APP))
+
+$(SOTNSTR_APP): tools/sotn_str/Cargo.toml $(wildcard tools/sotn_str/src/*)
+	cargo build --release --manifest-path tools/sotn_str/Cargo.toml
+$(SOTNDISK): $(GO) $(wildcard $(SOTNDISK_DIR)/*.go)
+	cd $(SOTNDISK_DIR) && $(GO) build
+$(SOTNASSETS): $(GO) $(wildcard $(SOTNASSETS_DIR)/*.go)
+	cd $(SOTNASSETS_DIR) && $(GO) build
+
+# Since venv is newly created, it can be reasonably assumed that the python requirements need to be installed
+$(VENV_DIR):
+	$(call echo,Creating python virtual environment) $(SYSTEM_PYTHON) -m venv $(VENV_DIR)
+	$(MAKE) python-dependencies
+tools/python-dependencies.make.chkpt: tools/requirements-python.txt | $(VENV_DIR)
+	$(PIP) install -r tools/requirements-python.txt && touch $@
+python-dependencies: tools/python-dependencies.make.chkpt
 
-define list_o_files_psp
-	$(foreach file,$(call list_src_files_psp,$(1)),$(BUILD_DIR)/$(file).o)
-endef
-### End old header ###
+tools/graphviz.make.chkpt: tools/python-dependencies.make.chkpt
+	sudo apt update && sudo apt-get install -y graphviz && touch $@
+graphviz: tools/graphviz.make.chkpt
 
-ifneq (,$(filter $(VERSION),us hd)) # Both us and hd versions use the PSX platform
+$(GO):
+	curl -sSfL -O https://go.dev/dl/go1.22.4.$(OS)-$(ARCH).tar.gz
+	$(muffle)tar -C $(HOME) -xzf go1.22.4.$(OS)-$(ARCH).tar.gz; rm go1.22.4.$(OS)-$(ARCH).tar.gz
+$(BIN_DIR)/%.tar.gz: $(BIN_DIR)/%.tar.gz.sha256
+	curl -sSfL -o $@ https://github.com/Xeeynamo/sotn-decomp/releases/download/cc1-psx-26/$*.tar.gz
+$(BIN_DIR)/%: $(BIN_DIR)/%.tar.gz
+	$(muffle)sha256sum --check $<.sha256
+	$(muffle)cd $(BIN_DIR) && tar -xzf $(notdir $<); rm $(notdir $<)
+	$(muffle)touch $@
+
+ifneq ($(filter $(VERSION),us hd),) # Both us and hd versions use the PSX platform
 include Makefile.psx.mk
 else ifeq ($(VERSION),pspeu)
 include Makefile.psp.mk
 else ifeq ($(VERSION),saturn)
 include Makefile.saturn.mk
 endif
 
+# this help target will find targets which are followed by a comment beginning with '#' '#' '@' and
+# print them in a summary form. Any comments on a line by themselves with start with `#' '#' '@'
+# will act as section dividers.
+help:
+	printf "\nUsage: make [VERSION=version] <target> …\n"
+	grep -F -h "##@" $(MAKEFILE_LIST) | grep -F -v "grep -F" | sed -e 's/\\$$//' | awk 'BEGIN {FS = ":*[[:space:]]*##@[[:space:]]?"}; \
+	{ \
+		if($$2 == "") \
+			printf ""; \
+		else if($$0 ~ /^#/) \
+			printf "\n%s\n", $$2; \
+		else if($$1 == "") \
+			printf "     %-20s%s\n", "", $$2; \
+		else { \
+            system("tput setaf 4; printf \"\n    %-20s \" " $$1 "; tput sgr0"); \
+            printf "%s\n", $$2; \
+        }; \
+	}'
+
```

Comment:
No functional change here, just moved things so that all of the help text is in one location.  While working through all of this, mainly at the beginning, I felt as though having the help text interspersed throughout the targets was more hindrance than help.  I didn't review to see if anything needed to be added, so there are probably some things that need to be adjusted with the help text.